### PR TITLE
Fix/33 collected fixes and usage improvements

### LINKED ISSUE #33: Collected fixes and usage improvements
- Fix - debug regime as default is not expected
- Fix - add missing cache resolving when response action performed

### PR DESCRIPTION
**Pull request by miroslavpojer**
Fix - debug regime as default is not expected.
Fix - add missing cache resolving when response action performed.

Closes #33

### CODE REVIEW

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala [hunk 1]
```diff
@@ -59,12 +59,13 @@ object Response {
...
    *
+   * @param environment The current environment.
    * @param response   The response on which actions will be performed.
-   * @param responseAction The set of response actions that dictate what actions will be performed on the response.
+   * @param responseActions The set of response actions that dictate what actions will be performed on the response.
    * @return           Boolean indicating whether all response actions passed (true) or any response action failed (false). TODO
    * @throws IllegalArgumentException If an response action group is not supported.
    */
-  def perform(response: Response, responseAction: Seq[ResponseAction]): Try[Unit] = {
+  def perform(environment: Environment, response: Response, responseActions: Seq[ResponseAction]): Try[Unit] = {
```
Comment (dk1844) [inline testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala:68]: I don't see the `environment` being used anywhere.

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala [hunk 2]
```diff
@@ -80,15 +81,15 @@ object Response {
       Logger.debug(s"Response-${resolvedResponseAction.group}: '${resolvedResponseAction.name}' - error details:$baseLog$exceptionLog")
     }
 
-    responseAction.iterator.map { assertion =>
-      val resolvedResponseAction: ResponseAction = assertion.resolveByRuntimeCache()
+    responseActions.iterator.map { responseAction =>
```
Comment (dk1844) [inline testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala:84]: Why not just

```suggestion
    responseActions.map { responseAction =>
```

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala [hunk 3]
```diff
@@ -59,12 +59,13 @@ object Response {
...
    * @param response   The response on which actions will be performed.
-   * @param responseAction The set of response actions that dictate what actions will be performed on the response.
+   * @param responseActions The set of response actions that dictate what actions will be performed on the response.
    * @return           Boolean indicating whether all response actions passed (true) or any response action failed (false). TODO
    * @throws IllegalArgumentException If an response action group is not supported.
    */
-  def perform(response: Response, responseAction: Seq[ResponseAction]): Try[Unit] = {
+  def perform(environment: Environment, response: Response, responseActions: Seq[ResponseAction]): Try[Unit] = {
     def logParameters(response: Response, resolvedResponseAction: ResponseAction, exception: Option[Throwable] = None): Unit = {
       val filteredParams = resolvedResponseAction.params.filter(_._1 != "method").map { case (k, v) => s"$k->$v" }.mkString(", ")
```
Comment (dk1844) [inline testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala:70]: For readability, I would probably change this to
```suggestion
      val filteredParams = resolvedResponseAction.params
        .filter { case (k, _) => k != "method"}
        .map { case (k, v) => s"$k->$v" }
        .mkString(", ")
```
.

The possible alternative of 
```suggestion
      val filteredParams = resolvedResponseAction.params
        .flatMap {
          case (k, v) if k != "method" => Some(s"$k->$v")
          case _ => None
        }
        .mkString(", ")
```
may be too convoluted.

File: testApi/src/main/scala/africa/absa/testing/scapi/logging/Logger.scala [hunk 4]
```diff
@@ -25,13 +25,16 @@ object Logger {
...
-      .map(_.getClassName)
-      .getOrElse("unknown")
-    val logger = LogManager.getLogger(callerClassName)
-    if (logger.isEnabled(level)) logger.log(level, message)
+  private def log(messageLevel: Level, message: String): Unit = {
+    if (messageLevel.isMoreSpecificThan(level)) {
+      val callerClassName = new Exception().getStackTrace.drop(1)
+        .find(_.getClassName != getClass.getName)
+        .map(_.getClassName)
+        .getOrElse("unknown")
```
Comment (dk1844) [inline testApi/src/main/scala/africa/absa/testing/scapi/logging/Logger.scala:33]: This is quite creative 😆

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala (hunk 1 above)
Comment (miroslavpojer) [reply testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala]: Fixed in commit - a446cfff190bbb995b33cfc8b677cfba6bff48cd.
It was not cleaned part from previous development.

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala (hunk 2 above)
Comment (miroslavpojer) [reply testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala]: Fixed in commit a446cfff190bbb995b33cfc8b677cfba6bff48cd.

File: testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala (hunk 3 above)
Comment (miroslavpojer) [reply testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala]: Used more functional solution with flatmap. See in commit a446cfff190bbb995b33cfc8b677cfba6bff48cd.

**Review by dk1844 — APPROVED** [review]

**Pull request merged**